src/lightning/data/streaming/cache.py

```diff
@@ -13,32 +13,33 @@
 
 import logging
 import os
-from typing import Any, Dict, List, Literal, Optional, Tuple, Union
+from typing import Any, Dict, List, Optional, Tuple, Union
 
 from lightning.data.datasets.env import _DistributedEnv
 from lightning.data.streaming.constants import (
     _INDEX_FILENAME,
-    _LIGHTNING_CLOUD_GREATER_EQUAL_0_5_46,
+    _LIGHTNING_CLOUD_GREATER_EQUAL_0_5_47,
     _TORCH_GREATER_EQUAL_2_1_0,
 )
 from lightning.data.streaming.item_loader import BaseItemLoader
 from lightning.data.streaming.reader import BinaryReader
 from lightning.data.streaming.sampler import ChunkedIndex
 from lightning.data.streaming.writer import BinaryWriter
 
-if _LIGHTNING_CLOUD_GREATER_EQUAL_0_5_46:
-    from lightning_cloud.resolver import _find_remote_dir, _try_create_cache_dir
-
 logger = logging.Logger(__name__)
 
+if _LIGHTNING_CLOUD_GREATER_EQUAL_0_5_47:
+    from lightning_cloud.resolver import Dir, _resolve_dir
+else:
+
+    class Dir:
+        pass
```

Comment:
Can't we set lightning-cloud 0.5.47 as the minimum version?

Reply:
done !